openssl/openssl_port_hmac.c

```diff
@@ -14,7 +14,12 @@ DEFINEFUNCINTERNAL(EVP_PKEY *,
 		   (int type, ENGINE *e, const unsigned char *key, int keylen),
 		   (type, e, key, keylen))
 DEFINEFUNCINTERNAL(int, EVP_MD_CTX_reset, (EVP_MD_CTX *ctx), (ctx))
+
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
```

Comment:
Is that a magic number we can extract as a global named const?

Reply:
you mean somethign like:
```
#define OPENSSL_3_0_VERSION 0x30000000L
#if OPENSSL_VERSION_NUMBER >= OPENSSL_3_0_VERSION
```
?

Reply:
Maybe we could add a compatibility definition of `OPENSSL_VERSION_PREREQ` (added in 1.1.1) so we can check the version with `#if OPENSSL_VERSION_REREQ(3,0) ...` ?

Reply:
I think using a macro or constant could be more intuitive given that we only discern between a handful of openssl versions.  But let's merge this and start a follow up conversation given the existing precedent for using numeric literals.

Reply:
Extracted to a new issue #151